test/extended/machines/machines.go

```diff
@@ -98,6 +98,61 @@ var _ = g.Describe("[Feature:Machines][Smoke] Managed cluster should", func() {
 			e2e.Failf("Machine resources missing for nodes: %s", strings.Join(nodeNames.List(), ", "))
 		}
 	})
+
+	g.It("not have machine resources with provider APIVersion in k8s.io namespace", func() {
+		cfg, err := e2e.LoadConfig()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		c, err := e2e.LoadClientset()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		dc, err := dynamic.NewForConfig(cfg)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("checking for the openshift machine api operator")
+		// TODO: skip if platform != aws
+		skipUnlessMachineAPIOperator(c.CoreV1().Namespaces())
+
+		g.By("ensuring every machine provider APIVersion is in the right namespace")
+		machineClient := dc.Resource(schema.GroupVersionResource{Group: "machine.openshift.io", Resource: "machines", Version: "v1beta1"})
+		var lastMachines []objx.Map
+		if err := wait.PollImmediate(3*time.Second, operatorWait, func() (bool, error) {
+			obj, err := machineClient.List(metav1.ListOptions{})
+			if err != nil {
+				e2e.Logf("Unable to check for machines: %v", err)
+				return false, nil
+			}
+			machines := objx.Map(obj.UnstructuredContent())
+			items := objects(machines.Get("items"))
+			if len(items) == 0 {
+				e2e.Logf("No machine objects found")
+				return true, nil
+			}
+
+			for _, machine := range items {
+				providerAPIVersion := providerAPIVersionFromMachine(machine)
+				e2e.Logf(providerAPIVersion)
+				if strings.Contains(providerAPIVersion, "k8s.io") {
```

Comment:
Can we go with `!strings.Contains(providerAPIVersion, "openshift.io")` unless we plan to also allow groups other than `openshift.io`.

Reply:
I think one of the metal providers isn't running in openshift.io namespace. Hence, I went with the not in k8s.io logic.

Reply:
In that case I would still go with white list rather than black list. We can always extend the white list later. Btw. what's the metal provider group? @cynepco3hahue do you know?

Reply:
`metal3.io`